### DISCUSSION TITLE
Nulleinspeisung - Dynamic Power limiter Settings

### DISCUSSION THREAD

**Discussion by olomberg** [hoylabs/OpenDTU-OnBattery#180] [Q&A] [answered]
Guten Morgen.

Habe seid einigen Tagen meinem HM-1500 mit den Settings für die Nulleinspeisung laufen. Funktioniert an sich auch soweit alles. alle Daten werden bei mir in eine InfluxDB geschrieben, welche ich mit Grafana auswerte. Dabei ist mit folgendes aufgefallen. Über den Tag werden immer noch ca. 30-100 Wh aus dem Netz gezogen, obwohl genügend Sonne vorhanden ist. Dazu auch die Grafik. Gelb = Solar, Blau = Gridbezug
![image](https://user-images.githubusercontent.com/99411925/232985047-814dec04-11d1-4dd6-9d7b-e1ad5cb9e40b.png)

Meine Vermutung ist an dieser Stelle ein Delta zwischen der Leistung und dem Limit. Wenn das Limit z.B. auf 200W gesetzt wird kommen nicht immer 200W rein. Trotz ausreichender Sonne. Hier auch dazu ein Ausschnitt aus Influx, welcher das Delta zeigt.

![image](https://user-images.githubusercontent.com/99411925/232986139-f67ad74d-a1cf-4e64-a86d-215394c09370.png)

Aus meiner Sicht könnte man das Problem einfach beheben.
Alternative 1: Neues Feld, welches in die Berechnung aufgenommen wird. Korrekturfaktor, o.A.
Alternative 2: Möglichkeit zu Eingabe eines negativen "Erlaubter Stromverbrauch". Aktuell lassen sich nur Werte >= 0 eingeben.

Wäre mir egal, wenn ich ein wenig in's Netz einspeise. Geht mir einfach um die Vermeidung von Strombezug. Würde bei Alternative 2 den Wert auf "-100" stellen.

Frage 1: Liegt es wirklich an de Delta?
Frage 2: Könnte mann dieses ändern.

Vielen Dank
Gruss
Oliver

**Comment by helgeerbe** [comment, marked as answer]
Habe ich schon auf dem Plan #171 Würde einen negativen Wert zulassen.

**Reply by olomberg** [reply]
Top. Danke. Aktueller Workaround: Backup der Config, Edit in "config.json", Import der Config.

**Reply by helgeerbe** [reply]
Danke für die die Bestätigung. Dann brauche ich tatsächlich in der Oberfläche nur die Eingabeüberprüfung anpassen.

**Reply by olomberg** [reply]
Moin.

Bin auf Version 41cc5ff. Kann jetzt einen negativen Wert beim "Angestrebten Netzbezug" eingeben und speichern. Der Wert wird aber nicht wirklich gespeichert und übernommen. Könnt ihr das bitte nochmals checken. Danke.
Gruss
Oliver

**Reply by helgeerbe** [reply]
Danke für den Hinweis. Sollte jetzt mit 38b990fbbc4a1025a9f0cb8ffd9f227d0a293f19 gefixt sein. Da wurde auf den falschen Typ uint32_t statt int32_t  gecasted. 

Zumindest das Speicher des Wertes funktioniert jetzt. Testen konnte ich noch nicht, da ich bei der Dunkelheit keinen Export mehr hinbekomme.

**Comment by mawa85** [comment]
Kurze Verständnisfrage: Kann ich mit der Version eine Nulleinspeisung (nur grob natürlich, die Limitierungen sind mir bekannt) realisieren, rein als PV Inverter ohne Batterie? Die aktuellen Verbrauchswerte vom Zähler liegen mir bereits per MQTT vor

**Reply by helgeerbe** [reply]
Ich glaube die Diskussion hatten wir schon mal in #179  Hoffe, das hilft weiter.
